### PR TITLE
Fix LightmapGI taking editor-only and sky-only lights into account

### PR DESCRIPTION
**Pull request by Calinou**
These lights are not visible when running the project (or at all), so they shouldn't affect the lightmap baking process.

VoxelGI already takes care of sky-only lights, but not editor only lights (I've looked and didn't find a way to implement it).

- See https://github.com/godotengine/godot/issues/82834#issuecomment-1774708084.

### CODE REVIEW

**Comment by jcostello** [comment]
Nice fix, this help me to not disable the light mode before baking :)

File: doc/classes/Light3D.xml [hunk 1]
```diff
@@ -52,7 +52,7 @@
 			[b]Note:[/b] Only effective for [OmniLight3D] and [SpotLight3D], and only when [member shadow_enabled] is [code]true[/code].
 		</member>
 		<member name="editor_only" type="bool" setter="set_editor_only" getter="is_editor_only" default="false">
-			If [code]true[/code], the light only appears in the editor and will not be visible at runtime.
+			If [code]true[/code], the light only appears in the editor and will not be visible at runtime. If [code]true[/code], the light will never be baked in [LightmapGI] regardless of its [member light_bake_mode].
```
Comment (akien-mga) [inline doc/classes/Light3D.xml:55]: Repeating "if true" sounds repetitive. How about "Additionally,"?

**Review by clayjohn — APPROVED** [review]
Makes sense to me

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**